pkg/controller/operands/namespace.go

```diff
@@ -0,0 +1,108 @@
+package operands
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/pkg/apis/hco/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/controller/common"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+// Handles a specific resource (a CR, a configMap and so on), to be run during reconciliation
+type namespaceHandler struct {
+	// K8s client
+	Client client.Client
+	Scheme *runtime.Scheme
+}
+
+func newNamespaceHandler(Client client.Client, Scheme *runtime.Scheme) *namespaceHandler {
+	return &namespaceHandler{
+		Client: Client,
+		Scheme: Scheme,
+	}
+}
+
+func (h *namespaceHandler) ensure(req *common.HcoRequest) *EnsureResult {
+	required := NewNamespace(req.Instance)
+	key := client.ObjectKeyFromObject(required)
+	found := &corev1.Namespace{}
+	err := h.Client.Get(req.Ctx, key, found)
+	if err != nil {
+		req.Logger.Error(err, "failed fetching namespace")
+		return &EnsureResult{
+			Err: err,
+		}
+	}
+
+	res := NewEnsureResult(found)
+	res.SetName(key.Name)
+
+	needUpdate := false
+	if found.Labels == nil {
+		found.Labels = make(map[string]string)
+	}
+	if found.Annotations == nil {
+		found.Annotations = make(map[string]string)
+	}
+	for k, v := range required.Labels {
+		if found_v, hasKey := found.Labels[k]; hasKey {
+			if found_v != v {
+				found.Labels[k] = v
+				needUpdate = true
+			}
+		} else {
+			found.Labels[k] = v
+			needUpdate = true
+		}
+	}
+	for k, v := range required.Annotations {
+		if found_v, hasKey := found.Annotations[k]; hasKey {
+			if found_v != v {
+				found.Annotations[k] = v
+				needUpdate = true
+			}
+		} else {
+			found.Annotations[k] = v
+			needUpdate = true
+		}
+	}
+
+	if needUpdate {
+		if req.HCOTriggered {
+			req.Logger.Info("Updating existing namespace to new opinionated values")
+		} else {
+			req.Logger.Info("Reconciling an externally updated namespace to its opinionated values")
+		}
+		err := h.Client.Update(req.Ctx, found)
+		if err != nil {
+			if err != nil {
+				req.Logger.Error(err, "failed updating the namespace")
+				return &EnsureResult{
+					Err: err,
+				}
+			}
+		}
+		res.SetUpdated()
+		res.SetOverwritten(!req.HCOTriggered)
+		return res
+	}
+	return res.SetUpgradeDone(req.ComponentUpgradeInProgress)
+}
+
+func (h namespaceHandler) reset( /* No implementation */ ) {}
+
+// NewNamespace creates namespace definition
+func NewNamespace(hc *hcov1beta1.HyperConverged) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   hc.Namespace,
+			Labels: getLabels(hc, hcoutil.AppComponentDeployment),
```

Comment:
`getLabels` adds the `app.kubernetes.io/managed-by` label - do we want it on the namespace?

Reply:
Yes, I agree: we are only setting an annotation, `app.kubernetes.io/managed-by` is too much.